skyportal/handlers/photometry.py

```diff
@@ -68,6 +122,33 @@ def post(self):
             DBSession().add(p)
             DBSession().flush()
             ids.append(p.id)
+        if 'thumbnails' in data:
+            for thumb in data['thumbnails']:
+                basedir = Path(os.path.dirname(__file__))/'..'/'..'
+                if os.path.abspath(basedir).endswith('skyportal/skyportal'):
+                    basedir = basedir/'..'
+                file_uri = os.path.abspath(
+                    basedir/f'static/thumbnails/{source.id}_{thumb["ttype"]}.png')
+                if not os.path.exists(os.path.dirname(file_uri)):
+                    (basedir/'static/thumbnails').mkdir(parents=True)
+                file_bytes = base64.b64decode(thumb['data'])
+                im = Image.open(io.BytesIO(file_bytes))
+                if im.format != 'PNG':
+                    return self.error('Invalid thumbnail image type. Only PNG are supported.')
+                if not (100, 100) <= im.size <= (500, 500):
+                    return self.error('Invalid thumbnail size. Only thumbnails '
+                                      'between (100, 100) and (500, 500) allowed.')
+                try:
+                    t = Thumbnail(type=thumb['ttype'],
+                                  photometry_id=ids[0],
+                                  file_uri=file_uri,
+                                  public_url=f'/static/thumbnails/{source.id}_{thumb["ttype"]}.png')
+                    DBSession.add(t)
+                except TypeError:
+                    return self.error('Invalid thumbnail type. Please refer to '
+                                      'API docs for a list of allowed types.')
+                with open(file_uri, 'wb') as f:
```

Comment:
This logic is incorrect: do not write file if DB call fails.

Reply:
No, if the DB call fails, it returns immediately.

Reply:
Well, if it returns a TypeError.  But who knows what all else can go wrong.  So we should probably include a catch-all as well.

Reply:
Sure thing -- done!